lib/authorizer/oci-v1.js

```diff
@@ -0,0 +1,292 @@
+    HEADER_AUTHORIZATION = 'authorization',
+    HEADER_DATE = 'x-date',
+    HEADER_HOST = 'host',
+    CONTENT_TYPE_JSON = 'application/json',
+    EXTRA_HEADER_METHODS = ['POST', 'PUT', 'PATCH'],
+    PSEUDOHEADER_REQUEST_TARGET = '(request-target)';
+
+// --------------------  POLYFILL for input to httpSignature.sign ---------------------------
+
+class SignerRequest {
+    constructor (method, url, headers) {
+        this.headers = headers;
+        this.method = method;
+        this.path = UrlParser.parse(url).path;
+    }
+
+    getHeader (name) {
+        return this.headers.get(name);
+    }
+
+    setHeader (name, value) {
+        this.headers.set(name, value);
+    }
+}
+
+function computeBodyHash (body, algorithm, digestEncoding, callback) {
+    if (!(body && algorithm && digestEncoding) || body.isEmpty()) { return callback(); }
+
+    var hash = crypto.createHash(algorithm),
+        originalReadStream,
+        rawBody,
+        urlencodedBody,
+        graphqlBody;
+
+
+    if (body.mode === RequestBody.MODES.raw) {
+        rawBody = bodyBuilder.raw(body.raw).body;
+        hash.update(rawBody);
+
+        // hash.update('\n');
+        return callback(hash.digest(digestEncoding), rawBody.length);
+    }
+
+    if (body.mode === RequestBody.MODES.urlencoded) {
+        urlencodedBody = bodyBuilder.urlencoded(body.urlencoded).form;
+        urlencodedBody = urlEncoder.encodeQueryString(urlencodedBody);
+        hash.update(urlencodedBody);
+        hash.update('\n');
+
+        return callback(hash.digest(digestEncoding), urlencodedBody.length + 1);
+    }
+
+    if (body.mode === RequestBody.MODES.file) {
+        originalReadStream = _.get(body, 'file.content');
+
+        if (!originalReadStream) {
+            return callback();
+        }
+
+        return originalReadStream.cloneReadStream(function (err, clonedStream) {
+            if (err) { return callback(); }
+            var streamContentLength = 0;
+
+            clonedStream.on('data', function (chunk) {
+                hash.update(chunk);
+                streamContentLength += chunk.length;
+            });
+
+            clonedStream.on('end', function () {
+                hash.update('\n');
+                callback(hash.digest(digestEncoding), streamContentLength);
+            });
+        });
+    }
+
+    if (body.mode === RequestBody.MODES.graphql) {
+        graphqlBody = bodyBuilder.graphql(body.graphql).body;
+        hash.update(graphqlBody);
+        hash.update('\n');
+
+        return callback(hash.digest(digestEncoding), graphqlBody.length + 1);
+    }
+
+    // @todo: Figure out a way to calculate hash for formdata body type.
+
+    // ensure that callback is called if body.mode doesn't match with any of the above modes
+    return callback();
+}
+// ------------------------------------------------------------
+
+// eslint-disable-next-line one-var
+var convertToDomHeadersObject = function (postmanHeaders) {
```

Comment:
Why do we have to convert `PostmanHeaderList`?
Facing any issue with the get/set methods exposed by it?